pkg/sfu/bwe/sendsidebwe/send_side_bwe.go

```diff
@@ -93,14 +93,7 @@ func (s *SendSideBWE) SetBWEListener(bweListener bwe.BWEListener) {
 }
 
 func (s *SendSideBWE) Reset() {
-	s.congestionDetector = newCongestionDetector(congestionDetectorParams{
-		Config: s.params.Config.CongestionDetector,
-		Logger: s.params.Logger,
-	})
```

Comment:
This is where the leaking goworker happened. There were two issues due to just replacing congestion detector with new instance
- old instance's goroutine was leaked
- BWE listener was not set on the new instance, so congestion notifications went nowhere.